core/cap-0067.md

```diff
@@ -145,14 +151,24 @@ contract: asset, topics: ["clawback", from:Address, sep0011_asset:String], data:
+### Update the `transfer` function to take in a `MuxedAccount` for `from` and `to`
+
+This change will allow the `transfer` event to emit muxed information for exchange support. 
+
+### Update the `data` field in the `transfer` event from an integer to a map to store additional memo information.
 
-Add the following function to the Stellar Asset Contract
-`transfer_with_memo(from: Address, to: Address, amount: i128, from_muxed_id: Option<u64>, to_muxed_id: Option<u64>)` and extend the `transfer` event topics to `["transfer", from: Address, to: Address, sep0011_asset: String, muxed_ids:Vec<Option<U64>>]`.
+Make the `data` field a map on the `transfer` event. Here are the following relevant scenarios and what the resulting events should look like - 
 
-The `transfer_with_memo` function will forward the muxed id's into the new muxed_ids vector on the `transfer` event. The first element will be the muxed id for `from`, and the second element will be the muxed id for `to`. If the optional muxed id is not specified, then a `SCV_VOID` `SCVal` will be emitted instead. The muxed parameters must be passed to the `transfer_with_memo`'s `require_auth` call.
+1. For Soroban transfers -
+   1. If either or both `from` and `to` are of type `KEY_TYPE_MUXED_ED25519`, then the respective `u64` IDs will be pulled out and emitted in the data field - `topics: ["transfer", from:G, to:G] data: { amount: i128, from_muxed_id:u64, to_muxed_id:u64 }`. Note that `from` and `to` are the ed25519 addresses, not the muxed address.
+2. For any of the `transfer` events that can be emitted by non-soroban operations -
+   1. The memo in a non-Soroban transaction can be one of `MEMO_NONE`, `MEMO_TEXT`, `MEMO_ID`, `MEMO_HASH`, or `MEMO_RETURN`. This memo will be forwarded for the destination, so `to_muxed_id` can be one of those types mentioned eg. `topics: ["transfer", from:G, to:G] data: { amount: i128, from_muxed_id:u64, to_muxed_id:u64|String|BytesN<32> }`.
+   2. If the sender of the `transfer` is muxed, then `from_muxed_id` will be emitted with the `MEMO_ID` on that muxed account.
+   3. If the destination of the `transfer` is muxed, then `to_muxed_id` will be emitted with the `MEMO_ID` on the muxed account. If the destination is not muxed, but the transaction memo is set, then `to_muxed_id` will be emitted with the memo on the transaction.
+3. For both scenarios, note that the map will only contain a `from_muxed_id` and/or `to_muxed_id` if there is muxed information for the addresses, so the data field can have one, two, or three elements.
 
-Note that when the `transfer` event change is done for the token standard, we should take into account the fact that the SAC emits `sep0011_asset` as a topic, which the token standard doesn't. To keep the indexes of the topics consistent, we should use a `SCV_VOID` topic as a placeholder for the `sep0011_asset` topic in the token standard. 
+### Prohibit the transaction memo from being set on Soroban transactions
```

Comment:
Do we care that a soroban transaction with `SOROBAN_CREDENTIALS_ADDRESS` and a muxed source account is not prohibited here?

Reply:
It seems safer to prohibit this as well. The muxed source should be pushed into the operation itself. I don't think it should matter that the muxed account has paid fees.

Reply:
done